pkg/agent/batch_remote_write_client.go

```diff
@@ -83,13 +110,15 @@ func (b *BatchWriteClient) batch(ctx context.Context) error {
 	expbackOff.MaxElapsedTime = b.writeInterval         // TODO: Subtract ~10% of interval to account for overhead in loop
 	expbackOff.InitialInterval = 500 * time.Millisecond // Let's not retry to aggressively to start with.
 
+	start := time.Now()
```

Comment:
Why not capture the start at the beginning of the method? Even before the locks. It would be semantically more correct, I believe.